runhouse/resources/serverless/aws_lambda_function.py

```diff
@@ -784,57 +829,107 @@ def aws_lambda_function(
         >>> lambdas_func = rh.aws_lambda_function(fn=summer, name="lambdas_func")
 
     """
-    if not any([name, paths_to_code, handler_function_name, runtime, fn, args_names]):
-        logger.error(
-            "Runhouse can't create a Lambda function. "
-            + "Please provide lambda's name and/or paths_to_code, handler_function_name, runtime and args_names"
-        )
-        raise Exception("NoEnoughArgsProvided")
-
     if name and not any(
         [paths_to_code, handler_function_name, runtime, fn, args_names]
     ):
         # Try reloading existing function
         return AWSLambdaFunction.from_name(name=name)
 
-    # ------- arguments validation -------
-    # TODO: use error msg inside the exception
-    if paths_to_code is None or len(paths_to_code) == 0:
-        logger.error("Please provide a path to the lambda handler file.")
-        raise RuntimeError
-    if handler_function_name is None or len(handler_function_name) == 0:
-        logger.error(
-            "Please provide the name of the function that should be executed by the lambda."
-        )
-        raise RuntimeError
-    if runtime is None or runtime not in SUPPORTED_RUNTIMES:
-        logger.error(
-            f"Please provide a supported lambda runtime, should be one of the following: {SUPPORTED_RUNTIMES}"
-        )
-        raise RuntimeError
-    if args_names is None:
-        logger.error(
-            "Please provide the names of the arguments provided to handler function, in the order they are"
-            + " passed to the lambda function."
-        )
-        raise RuntimeError
-
     # TODO: [SB] in the next phase, maybe add the option to create func from git.
 
-    fn_pointers = (
-        Function._extract_pointers(fn, reqs=env or []) if callable(fn) else None
-    )
+    if env is not None:
+        if isinstance(env, Env):
+            env_vars = env.env_vars
+            if not isinstance(env_vars, dict) and (
+                len(env_vars) > 0 or env_vars is not None
+            ):
+                raise TypeError("AWS Lambda accepts env_vars of dict type only")
+        elif isinstance(env, dict):
+            reqs = env["reqs"]
+            if isinstance(reqs, str):
+                reqs = [reqs]
+            env_vars = env["env_vars"]
+            f_name = name or fn.__name__ or handler_function_name
+            env = Env(reqs=reqs, name=f"{f_name}_env", env_vars=env_vars)
+        else:
+            received_type = str(type(env)).removeprefix("<class '").removesuffix("'>")
```

Comment:
why would this happen?

Reply:
A rare edge case. The probability it'll happen is really low, so removing it for now. Will add later on if it be necessary (will be more clear after users will start using AWS Lambdas).